MVP

manolo opened the pull request:
Connected to #1 

<!-- Reviewable:start -->
[<img src="https://reviewable.io/review_button.svg" height="40" alt="Review on Reviewable"/>](https://reviewable.io/reviews/vaadin/vaadin-upload/25)
<!-- Reviewable:end -->

tehapo commented:
Review status: 0 of 22 files reviewed at latest revision, 4 unresolved discussions.

---

*[bower.json, line 9 \[r1\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/25#-KAAN0UT2c0tV54FfwvN:-KAAN0UT2c0tV54FfwvO:-70426057) ([raw file](https://github.com/vaadin/vaadin-upload/blob/8282a9df3437a8962be7241cf7cef660fcb7baf6/bower.json#L9)):*
What is this `moduleType`? We don't have anything similar in our other elements.

---

*[bower.json, line 20 \[r1\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/25#-KAAMwO6cY-4mLVITKZu:-KAAMwO6cY-4mLVITKZv:-1770086845) ([raw file](https://github.com/vaadin/vaadin-upload/blob/8282a9df3437a8962be7241cf7cef660fcb7baf6/bower.json#L20)):*
Should be removed. We do want to publish this to the bower repository.

---

*[vaadin-upload.html, line 166 \[r1\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/25#-KAALWBTtY5Vy8O6RPrc:-KAALWBTtY5Vy8O6RPrd:314303817) ([raw file](https://github.com/vaadin/vaadin-upload/blob/8282a9df3437a8962be7241cf7cef660fcb7baf6/vaadin-upload.html#L166)):*
I think it would be a good idea to have a line break after each property to visually associate the documentation comment to the property that follows.

---

*[vaadin-upload.html, line 430 \[r1\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/25#-KAAJWGPJGcQQIXX1Sja:-KAAJWGPJGcQQIXX1Sjb:-211873232) ([raw file](https://github.com/vaadin/vaadin-upload/blob/8282a9df3437a8962be7241cf7cef660fcb7baf6/vaadin-upload.html#L430)):*
Typo, should be "Forbidden".

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-upload/25)*
<!-- Sent from Reviewable.io -->

magi42 commented:
Reviewed 19 of 22 files at r1.
Review status: 19 of 22 files reviewed at latest revision, 5 unresolved discussions.

---

*[docs/overview.asciidoc, line 11 \[r1\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/25#-KAALCobF2RFeMq_omfb:-KAALCocSI919k2RAEya:1555301936) ([raw file](https://github.com/vaadin/vaadin-upload/blob/8282a9df3437a8962be7241cf7cef660fcb7baf6/docs/overview.asciidoc#L11)):*
Use `[vaadinelement]#vaadin-upload#`.

For other comments, I already reviewed this file for PR #6, but it got merged a bit too early.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-upload/25)*
<!-- Sent from Reviewable.io -->

tehapo commented:
Review status: 19 of 22 files reviewed at latest revision, 7 unresolved discussions.

---

*[vaadin-upload.html, line 368 \[r1\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/25#-KAAbHraYWQyLMoCS7gW:-KAAbHrbAkpNGkbgRi_r:-727962680) ([raw file](https://github.com/vaadin/vaadin-upload/blob/8282a9df3437a8962be7241cf7cef660fcb7baf6/vaadin-upload.html#L368)):*
Minor nitpick – `Date.now()` would be a bit cleaner.

---

*[vaadin-upload.html, line 376 \[r1\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/25#-KAAbN78zk91pDtDFP8f:-KAAbN78zk91pDtDFP8g:-727962680) ([raw file](https://github.com/vaadin/vaadin-upload/blob/8282a9df3437a8962be7241cf7cef660fcb7baf6/vaadin-upload.html#L376)):*
Minor nitpick – `Date.now()` would be a bit cleaner.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-upload/25)*
<!-- Sent from Reviewable.io -->

tomivirkki commented:
Review status: 19 of 22 files reviewed at latest revision, 14 unresolved discussions.

---

*[vaadin-upload.html, line 19 \[r1\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/25#-KAA_LuuIWM2Md6QufDB:-KAA_LuuIWM2Md6QufDC:878018072) ([raw file](https://github.com/vaadin/vaadin-upload/blob/8282a9df3437a8962be7241cf7cef660fcb7baf6/vaadin-upload.html#L19)):*
I don't recall seeing this feature in the planned MVP features. Feature creep?

---

*[vaadin-upload.html, line 92 \[r1\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/25#-KAAaiy_Z0B5GGdeFUj1:-KAAaiy_Z0B5GGdeFUj2:-279228036) ([raw file](https://github.com/vaadin/vaadin-upload/blob/8282a9df3437a8962be7241cf7cef660fcb7baf6/vaadin-upload.html#L92)):*
There's lots of custom style properties here but they're missing documentation.

---

*[vaadin-upload.html, line 134 \[r1\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/25#-KAA_XSa-cCxavmmP5FY:-KAA_XSa-cCxavmmP5FZ:1110099759) ([raw file](https://github.com/vaadin/vaadin-upload/blob/8282a9df3437a8962be7241cf7cef660fcb7baf6/vaadin-upload.html#L134)):*
Why is there a content tag here? It allows you to place practically anything between the fileList and buttons....

---

*[vaadin-upload.html, line 136 \[r1\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/25#-KAA_tz8hgpJqQqtLNXt:-KAA_tz8hgpJqQqtLNXu:-1892111093) ([raw file](https://github.com/vaadin/vaadin-upload/blob/8282a9df3437a8962be7241cf7cef660fcb7baf6/vaadin-upload.html#L136)):*
Should this have no ink? The animation doesn't work in this case because it just jams right after clicking.

---

*[vaadin-upload.html, line 199 \[r1\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/25#-KAAbvRy3Nk8KhIo0gnn:-KAAbvRy3Nk8KhIo0gno:-1587953467) ([raw file](https://github.com/vaadin/vaadin-upload/blob/8282a9df3437a8962be7241cf7cef660fcb7baf6/vaadin-upload.html#L199)):*
This shouldn't be in the public api

---

*[vaadin-upload.html, line 207 \[r1\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/25#-KAAaxEeC1tAxv50QG9C:-KAAaxEeC1tAxv50QG9D:223028470) ([raw file](https://github.com/vaadin/vaadin-upload/blob/8282a9df3437a8962be7241cf7cef660fcb7baf6/vaadin-upload.html#L207)):*
cancelAll sounds like a function for canceling transmissions. Would showCancelAll be more descriptive?

---

*[vaadin-upload.html, line 224 \[r1\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/25#-KAAcdb7Inf4Vqwu1DBc:-KAAcdb7Inf4Vqwu1DBd:1805401595) ([raw file](https://github.com/vaadin/vaadin-upload/blob/8282a9df3437a8962be7241cf7cef660fcb7baf6/vaadin-upload.html#L224)):*
Why not noMultiple? I'm guessing it's because of "noink". Multiple is however a long word so camelCase would work better here. https://elements.polymer-project.org/elements/gold-email-input#property-noLabelFloat

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-upload/25)*
<!-- Sent from Reviewable.io -->

manolo commented:
Review status: 19 of 22 files reviewed at latest revision, 14 unresolved discussions.

---

*[bower.json, line 9 \[r1\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/25#-KAAN0UT2c0tV54FfwvN:-KAAeZjz0JsRN1ocu4yW:1356318061) ([raw file](https://github.com/vaadin/vaadin-upload/blob/8282a9df3437a8962be7241cf7cef660fcb7baf6/bower.json#L9)):*
https://github.com/bower/spec/pull/13/files

---

*[bower.json, line 20 \[r1\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/25#-KAAMwO6cY-4mLVITKZu:-KAAebp6PHSnl-EpdsnW:-499146009) ([raw file](https://github.com/vaadin/vaadin-upload/blob/8282a9df3437a8962be7241cf7cef660fcb7baf6/bower.json#L20)):*
Done.

---

*[docs/overview.asciidoc, line 11 \[r1\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/25#-KAALCobF2RFeMq_omfb:-KAAeotQkc3yp0dIPQ3j:-839002114) ([raw file](https://github.com/vaadin/vaadin-upload/blob/8282a9df3437a8962be7241cf7cef660fcb7baf6/docs/overview.asciidoc#L11)):*
Done in a separate PR

---

*[vaadin-upload.html, line 19 \[r1\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/25#-KAA_LuuIWM2Md6QufDB:-KAAezZ-HfUpGlgJM9Re:1024331042) ([raw file](https://github.com/vaadin/vaadin-upload/blob/8282a9df3437a8962be7241cf7cef660fcb7baf6/vaadin-upload.html#L19)):*
During the implementation we thought that it was better to have the file row in a separate component so we decided to do in this way. See it like a Bonus :-)

---

*[vaadin-upload.html, line 92 \[r1\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/25#-KAAaiy_Z0B5GGdeFUj1:-KAAfRO2Z-x8BHMzxTkP:1581119559) ([raw file](https://github.com/vaadin/vaadin-upload/blob/8282a9df3437a8962be7241cf7cef660fcb7baf6/vaadin-upload.html#L92)):*
Right should be documented at least in the API

---

*[vaadin-upload.html, line 134 \[r1\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/25#-KAA_XSa-cCxavmmP5FY:-KAAfdQ29tVzvw93wdEO:-1108154115) ([raw file](https://github.com/vaadin/vaadin-upload/blob/8282a9df3437a8962be7241cf7cef660fcb7baf6/vaadin-upload.html#L134)):*
Yep  you can place any message, icon etc. 

---

*[vaadin-upload.html, line 136 \[r1\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/25#-KAA_tz8hgpJqQqtLNXt:-KAAgwJYeLS95veh3aYy:-841596764) ([raw file](https://github.com/vaadin/vaadin-upload/blob/8282a9df3437a8962be7241cf7cef660fcb7baf6/vaadin-upload.html#L136)):*
@platosha can you say something?

---

*[vaadin-upload.html, line 166 \[r1\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/25#-KAALWBTtY5Vy8O6RPrc:-KAAdoNWaIWTHH62TdNe:-499146009) ([raw file](https://github.com/vaadin/vaadin-upload/blob/8282a9df3437a8962be7241cf7cef660fcb7baf6/vaadin-upload.html#L166)):*
Done.

---

*[vaadin-upload.html, line 199 \[r1\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/25#-KAAbvRy3Nk8KhIo0gnn:-KAAh1XYzOLx74q5c1G1:-921598442) ([raw file](https://github.com/vaadin/vaadin-upload/blob/8282a9df3437a8962be7241cf7cef660fcb7baf6/vaadin-upload.html#L199)):*
Agree, what about you @platosha ?

---

*[vaadin-upload.html, line 207 \[r1\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/25#-KAAaxEeC1tAxv50QG9C:-KAAgEBYCY3ytly9hEJp:-1115478056) ([raw file](https://github.com/vaadin/vaadin-upload/blob/8282a9df3437a8962be7241cf7cef660fcb7baf6/vaadin-upload.html#L207)):*
It actually aborts all the uploads, otherwise the uploading continues after the file element was removed from the dom, so cancel sounds good

---

*[vaadin-upload.html, line 224 \[r1\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/25#-KAAcdb7Inf4Vqwu1DBc:-KAAgd2NTUaZctthjqPD:706865435) ([raw file](https://github.com/vaadin/vaadin-upload/blob/8282a9df3437a8962be7241cf7cef660fcb7baf6/vaadin-upload.html#L224)):*
It was in the beginning but @platosha  convinced me that this is the tendency in Polymer

---

*[vaadin-upload.html, line 368 \[r1\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/25#-KAAbHraYWQyLMoCS7gW:-KAAduzEotk42XletCT5:-499146009) ([raw file](https://github.com/vaadin/vaadin-upload/blob/8282a9df3437a8962be7241cf7cef660fcb7baf6/vaadin-upload.html#L368)):*
Done.

---

*[vaadin-upload.html, line 376 \[r1\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/25#-KAAbN78zk91pDtDFP8f:-KAAdvKXfjRnvgGD88zj:-499146009) ([raw file](https://github.com/vaadin/vaadin-upload/blob/8282a9df3437a8962be7241cf7cef660fcb7baf6/vaadin-upload.html#L376)):*
Done.

---

*[vaadin-upload.html, line 430 \[r1\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/25#-KAAJWGPJGcQQIXX1Sja:-KAAe-Vu88rNRBQtcaJd:-499146009) ([raw file](https://github.com/vaadin/vaadin-upload/blob/8282a9df3437a8962be7241cf7cef660fcb7baf6/vaadin-upload.html#L430)):*
Done.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-upload/25)*
<!-- Sent from Reviewable.io -->

platosha commented:
Review status: 19 of 22 files reviewed at latest revision, 14 unresolved discussions.

---

*[vaadin-upload.html, line 136 \[r1\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/25#-KAA_tz8hgpJqQqtLNXt:-KAAijcAIOJ2uJI23T-W:-1998646191) ([raw file](https://github.com/vaadin/vaadin-upload/blob/8282a9df3437a8962be7241cf7cef660fcb7baf6/vaadin-upload.html#L136)):*
I can. No, this shouldn't have `noink`. The button ripple is still visible on mobile, and it's nice.

---

*[vaadin-upload.html, line 199 \[r1\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/25#-KAAbvRy3Nk8KhIo0gnn:-KAAhDHralRmiLK3lJ8b:4489115) ([raw file](https://github.com/vaadin/vaadin-upload/blob/8282a9df3437a8962be7241cf7cef660fcb7baf6/vaadin-upload.html#L199)):*
Disagree. This is used for switching the state of styles for `:host`. While styling is not an MVP feature, we still simply cannot prevent everyone to target the `:host` element from the outside with their own styles. In this case, their styles might get overridden by `dragover` state style. Hence we should also expose this state attribute as public and document it.

---

*[vaadin-upload.html, line 207 \[r1\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/25#-KAAaxEeC1tAxv50QG9C:-KAAiVvqExvkjYbg4vcU:-2120825894) ([raw file](https://github.com/vaadin/vaadin-upload/blob/8282a9df3437a8962be7241cf7cef660fcb7baf6/vaadin-upload.html#L207)):*
I would agree with `showCancelAll` name.

---

*[vaadin-upload.html, line 224 \[r1\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/25#-KAAcdb7Inf4Vqwu1DBc:-KAAgVsr1guzSebgEHvY:1856359952) ([raw file](https://github.com/vaadin/vaadin-upload/blob/8282a9df3437a8962be7241cf7cef660fcb7baf6/vaadin-upload.html#L224)):*
camelCase works for multiple words, but we have just a single word with no- here. There’s also HTML5 `<form novalidate/>` attr, see: https://developer.mozilla.org/en/docs/Web/HTML/Element/form#attr-novalidate

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-upload/25)*
<!-- Sent from Reviewable.io -->

platosha commented:
Reviewed 3 of 22 files at r1.
Review status: all files reviewed at latest revision, 14 unresolved discussions.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-upload/25)*
<!-- Sent from Reviewable.io -->

tehapo commented:
Review status: 19 of 22 files reviewed at latest revision, 14 unresolved discussions.

---

*[vaadin-upload.html, line 368 \[r1\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/25#-KAAbHraYWQyLMoCS7gW:-KAAzryWA-jhm8mXijHr:2007001995) ([raw file](https://github.com/vaadin/vaadin-upload/blob/8282a9df3437a8962be7241cf7cef660fcb7baf6/vaadin-upload.html#L368)):*
I actually meant the `now()` method directly in the `Date` prototype. So this line would be `var ini = Date.now();`

---

*[vaadin-upload.html, line 376 \[r1\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/25#-KAAbN78zk91pDtDFP8f:-KAAzfCwI99zRdlMzbz4:1593873906) ([raw file](https://github.com/vaadin/vaadin-upload/blob/8282a9df3437a8962be7241cf7cef660fcb7baf6/vaadin-upload.html#L376)):*
I actually meant the `now()` method directly in the `Date` prototype. So this line would be `last = Date.now();`

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-upload/25)*
<!-- Sent from Reviewable.io -->

manolo commented:
Review status: 19 of 22 files reviewed at latest revision, 14 unresolved discussions, some commit checks failed.

---

*[vaadin-upload.html, line 92 \[r1\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/25#-KAAaiy_Z0B5GGdeFUj1:-KAAoH-5UWa2VnkIYS2p:-499146009) ([raw file](https://github.com/vaadin/vaadin-upload/blob/8282a9df3437a8962be7241cf7cef660fcb7baf6/vaadin-upload.html#L92)):*
Done.

---

*[vaadin-upload.html, line 207 \[r1\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/25#-KAAaxEeC1tAxv50QG9C:-KAAo4bXOcPTIv_iofTv:-499146009) ([raw file](https://github.com/vaadin/vaadin-upload/blob/8282a9df3437a8962be7241cf7cef660fcb7baf6/vaadin-upload.html#L207)):*
Done.

---

*[vaadin-upload.html, line 368 \[r1\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/25#-KAAbHraYWQyLMoCS7gW:-KAB7VGf5stfttosPsfR:-1600038418) ([raw file](https://github.com/vaadin/vaadin-upload/blob/8282a9df3437a8962be7241cf7cef660fcb7baf6/vaadin-upload.html#L368)):*
I noticed when travis failed :-/ . Done

---

*[vaadin-upload.html, line 376 \[r1\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/25#-KAAbN78zk91pDtDFP8f:-KAB7fB1m7L7d55Vbez2:-499146009) ([raw file](https://github.com/vaadin/vaadin-upload/blob/8282a9df3437a8962be7241cf7cef660fcb7baf6/vaadin-upload.html#L376)):*
Done.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-upload/25)*
<!-- Sent from Reviewable.io -->

tomivirkki commented:
The upload doesn't seem to work on IE (tried the component demos)

---

Review status: 19 of 22 files reviewed at latest revision, 15 unresolved discussions, some commit checks failed.

---

*[vaadin-upload.html, line 19 \[r1\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/25#-KAA_LuuIWM2Md6QufDB:-KABKiYsragPrpIAtUNN:-676165249) ([raw file](https://github.com/vaadin/vaadin-upload/blob/8282a9df3437a8962be7241cf7cef660fcb7baf6/vaadin-upload.html#L19)):*
File row as a separate web component is definitely the way to go. But I don't think allowing user to provide a custom replacement is a planned feature for 1.0 (not to mention MVP alpha)

---

*[vaadin-upload.html, line 134 \[r1\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/25#-KAA_XSa-cCxavmmP5FY:-KAB98ZbvRxjpitnfZXL:-1084978374) ([raw file](https://github.com/vaadin/vaadin-upload/blob/8282a9df3437a8962be7241cf7cef660fcb7baf6/vaadin-upload.html#L134)):*
Hmm...at least the content should be targeted like ".drop-label" above. As it is now it'll pick up the possible .file template also. Was this a planned feature?

---

*[vaadin-upload.html, line 136 \[r1\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/25#-KAA_tz8hgpJqQqtLNXt:-KAAv8SNDd8WlHsxv10L:684735669) ([raw file](https://github.com/vaadin/vaadin-upload/blob/8282a9df3437a8962be7241cf7cef660fcb7baf6/vaadin-upload.html#L136)):*
The ripple is visible on desktop also (at least on my browsers) but it immediately jams on click giving a laggy impression

---

*[vaadin-upload.html, line 199 \[r1\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/25#-KAAbvRy3Nk8KhIo0gnn:-KAAvKCQpMYVqMzuwtv3:662242906) ([raw file](https://github.com/vaadin/vaadin-upload/blob/8282a9df3437a8962be7241cf7cef660fcb7baf6/vaadin-upload.html#L199)):*
You're talking about attributes but this is a public _porperty_ (that's reflected to an attribute)

---

*[vaadin-upload.html, line 224 \[r1\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/25#-KAAcdb7Inf4Vqwu1DBc:-KAAw1AgnRgfpfApVCp3:291601361) ([raw file](https://github.com/vaadin/vaadin-upload/blob/8282a9df3437a8962be7241cf7cef660fcb7baf6/vaadin-upload.html#L224)):*
Well, there's also a "formnovalidate" attribute where you linked so it's not a good reference....But if 'nofoo' is the practice with one-word properties in Polymer (?) then this is ok.

---

*[vaadin-upload.html, line 376 \[r1\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/25#-KAAbN78zk91pDtDFP8f:-KAB1m9yLF_YlEge2-hZ:2086991599) ([raw file](https://github.com/vaadin/vaadin-upload/blob/8282a9df3437a8962be7241cf7cef660fcb7baf6/vaadin-upload.html#L376)):*
Should be this._last = ... or something. It currently ends up in the global scope.

---

*[vaadin-upload.html, line 234 \[r3\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/25#-KABPOdO43ou9G4_fWko:-KABPOdO43ou9G4_fWkp:974409885) ([raw file](https://github.com/vaadin/vaadin-upload/blob/9dc0e1d623c1db066b08305d9e08aead289a035e/vaadin-upload.html#L234)):*
This should be private since files are pushed here only trough _addFile(). I assume pushing directly to this array isn't how it's planned to be used (it would also bypass all validation in _addFile)

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-upload/25#-:-KAB3o4ae5eiGzV6Xms1:-1961053281)*
<!-- Sent from Reviewable.io -->

Saulis commented:
Review status: 19 of 22 files reviewed at latest revision, 22 unresolved discussions, some commit checks failed.

---

*[vaadin-upload.html, line 137 \[r3\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/25#-KADvJh7c9Wr07ZHe6pU:-KADvJh7c9Wr07ZHe6pV:189000060) ([raw file](https://github.com/vaadin/vaadin-upload/blob/9dc0e1d623c1db066b08305d9e08aead289a035e/vaadin-upload.html#L137)):*
Should import `iron-icon` and `iron-icons` explicitly

---

*[vaadin-upload.html, line 153 \[r3\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/25#-KADvA3ZmocTWxjvSxqO:-KADvA3ZmocTWxjvSxqP:964691603) ([raw file](https://github.com/vaadin/vaadin-upload/blob/9dc0e1d623c1db066b08305d9e08aead289a035e/vaadin-upload.html#L153)):*
Should import ´paper-ripple` explicitly

---

*[vaadin-upload.html, line 401 \[r3\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/25#-KADvgvSHYqDY4IYGh77:-KADvgvSHYqDY4IYGh78:-1855034387) ([raw file](https://github.com/vaadin/vaadin-upload/blob/9dc0e1d623c1db066b08305d9e08aead289a035e/vaadin-upload.html#L401)):*
typo: elapsed

---

*[vaadin-upload.html, line 416 \[r3\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/25#-KADvqNdqcZI-HUvwJ6n:-KADvqNdqcZI-HUvwJ6o:1626867927) ([raw file](https://github.com/vaadin/vaadin-upload/blob/9dc0e1d623c1db066b08305d9e08aead289a035e/vaadin-upload.html#L416)):*
could all this status formattin be extracted to it's own function? `_uploadFile` is HUGE :hamburger: 

---

*[vaadin-upload.html, line 565 \[r3\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/25#-KADxKT7Z17fIt53ebDa:-KADxKT7Z17fIt53ebDb:152199897) ([raw file](https://github.com/vaadin/vaadin-upload/blob/9dc0e1d623c1db066b08305d9e08aead289a035e/vaadin-upload.html#L565)):*
Why not `Polymer.Base.fire(...)` ?

---

*[vaadin-upload.html, line 609 \[r3\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/25#-KADw9ablLY0pXLC2X1X:-KADw9acNQBOiqElOyOc:-1898803919) ([raw file](https://github.com/vaadin/vaadin-upload/blob/9dc0e1d623c1db066b08305d9e08aead289a035e/vaadin-upload.html#L609)):*
Would prefer `file-added`

---

*[vaadin-upload.html, line 619 \[r3\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/25#-KADwCmSyFYufHJUUxDm:-KADwCmT5K2E9pfgnZWt:1328484753) ([raw file](https://github.com/vaadin/vaadin-upload/blob/9dc0e1d623c1db066b08305d9e08aead289a035e/vaadin-upload.html#L619)):*
Would prefer `file-removed`

---

*[vaadin-upload.html, line 628 \[r3\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/25#-KADwlBuPcfzH_rQVILD:-KADwlBuPcfzH_rQVILE:1682421592) ([raw file](https://github.com/vaadin/vaadin-upload/blob/9dc0e1d623c1db066b08305d9e08aead289a035e/vaadin-upload.html#L628)):*
Would prefer something ending in "ed", `request-created` ?

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-upload/25)*
<!-- Sent from Reviewable.io -->

manolo commented:
Review status: 19 of 22 files reviewed at latest revision, 22 unresolved discussions, some commit checks failed.

---

*[vaadin-upload.html, line 609 \[r3\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/25#-KADw9ablLY0pXLC2X1X:-KAE0jynsKKxYZrYbrJ5:-135304822) ([raw file](https://github.com/vaadin/vaadin-upload/blob/9dc0e1d623c1db066b08305d9e08aead289a035e/vaadin-upload.html#L609)):*
We already had a long discussion about this. It seems that  it's more usual call events in present so as declarative attributes or handling functions are more meaningful: change, on-change, onChange, the same with click etc, in this case: file-add, on-file-add, onFileAdd 

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-upload/25)*
<!-- Sent from Reviewable.io -->

manolo commented:
Review status: 19 of 22 files reviewed at latest revision, 22 unresolved discussions, some commit checks failed.

---

*[vaadin-upload.html, line 619 \[r3\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/25#-KADwCmSyFYufHJUUxDm:-KAE1QAqZJmFpu9Bi5Xf:1252362392) ([raw file](https://github.com/vaadin/vaadin-upload/blob/9dc0e1d623c1db066b08305d9e08aead289a035e/vaadin-upload.html#L619)):*
same explication as above

---

*[vaadin-upload.html, line 628 \[r3\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/25#-KADwlBuPcfzH_rQVILD:-KAE1Sz700N1Zq6--zWh:678021867) ([raw file](https://github.com/vaadin/vaadin-upload/blob/9dc0e1d623c1db066b08305d9e08aead289a035e/vaadin-upload.html#L628)):*
same

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-upload/25)*
<!-- Sent from Reviewable.io -->

manolo commented:
All tests pass in IE but uploading does not start, since this is alpha1, will assume this defect and open a ticket so as we fix it before beta. WDYT?

---

Review status: 19 of 22 files reviewed at latest revision, 22 unresolved discussions, some commit checks failed.

---

*[vaadin-upload.html, line 19 \[r1\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/25#-KAA_LuuIWM2Md6QufDB:-KAEAWlboIY1Y3eFcflw:952242760) ([raw file](https://github.com/vaadin/vaadin-upload/blob/8282a9df3437a8962be7241cf7cef660fcb7baf6/vaadin-upload.html#L19)):*
But the feature is there, it's tested and documented, should we remove it, I don't see any reason.

---

*[vaadin-upload.html, line 136 \[r1\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/25#-KAA_tz8hgpJqQqtLNXt:-KAE6RxIWVJfOQlwpmxm:-678328240) ([raw file](https://github.com/vaadin/vaadin-upload/blob/8282a9df3437a8962be7241cf7cef660fcb7baf6/vaadin-upload.html#L136)):*
This happens because opening the file-selector should be done synchronous, and it's a native action which seems stoping everything in the JS until it's open. I don't see any solution but removing the noink.

---

*[vaadin-upload.html, line 199 \[r1\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/25#-KAAbvRy3Nk8KhIo0gnn:-KAE7Stxyt_OL-jiOIrk:-554026735) ([raw file](https://github.com/vaadin/vaadin-upload/blob/8282a9df3437a8962be7241cf7cef660fcb7baf6/vaadin-upload.html#L199)):*
If we want the property be used by users to customise their UI it should be named dragover not _dragover. I don't see a better way to have a private property and refect it to attribute but using setAttribute, etc which could seem not very Polymer-ish

---

*[vaadin-upload.html, line 153 \[r3\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/25#-KADvA3ZmocTWxjvSxqO:-KAE7DymOxE23He0idOc:-989565450) ([raw file](https://github.com/vaadin/vaadin-upload/blob/9dc0e1d623c1db066b08305d9e08aead289a035e/vaadin-upload.html#L153)):*
paper-button already does, I don't see the need any good reason?

---

*[vaadin-upload.html, line 565 \[r3\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/25#-KADxKT7Z17fIt53ebDa:-KAE2nzBo6IZBxZQ5Naz:1694859143) ([raw file](https://github.com/vaadin/vaadin-upload/blob/9dc0e1d623c1db066b08305d9e08aead289a035e/vaadin-upload.html#L565)):*
It doesn't work because the native component restrictions. But since fileInput is an input element it has the click() method, so I dont see any reason for not using it instead.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-upload/25#-:-KAE6-z18HH3E4yOEbmd:904937863)*
<!-- Sent from Reviewable.io -->

tomivirkki commented:
*[vaadin-upload.html, line 199 \[r1\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/25#-KAAbvRy3Nk8KhIo0gnn:-KAEB62gdCV07_B2dsb-:1493610006) ([raw file](https://github.com/vaadin/vaadin-upload/blob/8282a9df3437a8962be7241cf7cef660fcb7baf6/vaadin-upload.html#L199)):*
You can remove the property and use this.toggleAttribute('dragover', boolean);

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-upload/25)*
<!-- Sent from Reviewable.io -->

manolo commented:
Review status: 19 of 22 files reviewed at latest revision, 22 unresolved discussions, some commit checks failed.

---

*[vaadin-upload.html, line 134 \[r1\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/25#-KAA_XSa-cCxavmmP5FY:-KAEC5fJ90BPHbdbsawM:1453445559) ([raw file](https://github.com/vaadin/vaadin-upload/blob/8282a9df3437a8962be7241cf7cef660fcb7baf6/vaadin-upload.html#L134)):*
We wanted to leave things easy for the user who should not bother about classes.

---

*[vaadin-upload.html, line 199 \[r1\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/25#-KAAbvRy3Nk8KhIo0gnn:-KAEGKejgidrGnCYFf0w:-685411907) ([raw file](https://github.com/vaadin/vaadin-upload/blob/8282a9df3437a8962be7241cf7cef660fcb7baf6/vaadin-upload.html#L199)):*
Will leave this for a separate PR. Added TODO.

---

*[vaadin-upload.html, line 376 \[r1\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/25#-KAAbN78zk91pDtDFP8f:-KAEG2Nq3MybpiZ3iLjO:1986739258) ([raw file](https://github.com/vaadin/vaadin-upload/blob/8282a9df3437a8962be7241cf7cef660fcb7baf6/vaadin-upload.html#L376)):*
It was in the component scope before, and forgot to define again, now it's in the apprpriate scope. Done.

---

*[vaadin-upload.html, line 137 \[r3\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/25#-KADvJh7c9Wr07ZHe6pU:-KAEBonTAU3_aEyKEWxs:-1748325831) ([raw file](https://github.com/vaadin/vaadin-upload/blob/9dc0e1d623c1db066b08305d9e08aead289a035e/vaadin-upload.html#L137)):*
paper-button import everything. Anyway added iron-icons (it depends on iron-icon) and ripple

---

*[vaadin-upload.html, line 234 \[r3\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/25#-KABPOdO43ou9G4_fWko:-KAEDESNsYhOiCovTz3U:1536062093) ([raw file](https://github.com/vaadin/vaadin-upload/blob/9dc0e1d623c1db066b08305d9e08aead289a035e/vaadin-upload.html#L234)):*
Added a TODO for a separate PR, this change is big, and should be reviewed separately

---

*[vaadin-upload.html, line 401 \[r3\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/25#-KADvgvSHYqDY4IYGh77:-KAEGI25tWGCpKPsFYAx:-499146009) ([raw file](https://github.com/vaadin/vaadin-upload/blob/9dc0e1d623c1db066b08305d9e08aead289a035e/vaadin-upload.html#L401)):*
Done.

---

*[vaadin-upload.html, line 416 \[r3\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/25#-KADvqNdqcZI-HUvwJ6n:-KAEFzKEByWgN_JUXMck:-499146009) ([raw file](https://github.com/vaadin/vaadin-upload/blob/9dc0e1d623c1db066b08305d9e08aead289a035e/vaadin-upload.html#L416)):*
Done.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-upload/25)*
<!-- Sent from Reviewable.io -->

platosha commented:
Reviewed 3 of 6 files at r5, 1 of 6 files at r6.
Review status: 19 of 25 files reviewed at latest revision, 16 unresolved discussions.

---

*[vaadin-upload.html, line 234 \[r3\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/25#-KABPOdO43ou9G4_fWko:-KAEXAUz6mudCqRY96Bc:-1027903667) ([raw file](https://github.com/vaadin/vaadin-upload/blob/9dc0e1d623c1db066b08305d9e08aead289a035e/vaadin-upload.html#L234)):*
Pushing files directly to this array is not planned. But ability to watch for this array from the outside and collect all file names & related data is a very useful thing.

E. g., if used inside a form, we need to provide some values for file list. Exposing the files array for outside processing & value extraction is the simplest thing to do that.

I would rather leave the files array public with change notifications, perhaps read-only.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-upload/25)*
<!-- Sent from Reviewable.io -->
